build/common/installer/scripts/tomlparser.rb

```diff
@@ -162,8 +164,12 @@ def populateSettingValuesFromConfigMap(parsedConfig)
     #Get container logs route setting
     begin
       if !parsedConfig[:log_collection_settings][:route_container_logs].nil? && !parsedConfig[:log_collection_settings][:route_container_logs][:version].nil?
-        @containerLogsRoute = parsedConfig[:log_collection_settings][:route_container_logs][:version]
-        puts "config::Using config map setting for container logs route"
+        if !parsedConfig[:log_collection_settings][:route_container_logs][:version].empty?
```

Comment:
These checks are not needed. The way it used to work is - if the soute is not one of the valid values (including empty), it just uses default.

Reply:
since the default changed to v2 for Linux &  v1 for windows. In case of Linux, fallback option is v1. This is just safety check to  fall back to defaults if the customer puts as empty as version and also we dont collect the telemetry as empty string. Removing this will impact telemetry not matching with default.

Reply:
whats wrong if telemetry is empty ? your argument just doesn't make sense. telemetry should show whatever the customer puts there.

Reply:
what i am saying is, if customer has the cm with the version="" , we default to v2 but the telemetry will be "" since the telemetry uses the same version what's default.  do you see any issue adding !empty? check?